hack/get-machine-run-tests.sh

```diff
@@ -123,110 +150,146 @@ ROOT_VOLUME_SIZE=${ROOT_VOLUME_SIZE:-75}
+            echo keypair_name: "${OCT_KEYPAIR_NAME}" >> $HOME/.config/origin-ci-tool/aws_client_configuration.yml
         }
-    else
-        echo No such file "${OCT_PRIVATE_KEY_PATH:-}"
-        exit 1
     fi
+
+    oct provision remote all-in-one --os $OS --provider aws --stage build --name $INSTNAME
+
+    # based on
+    # https://github.com/openshift/aos-cd-jobs/blob/master/sjb/config/test_cases/test_branch_openshift_ansible_logging.yml
+    #  sync_repos:
+    #    - name: "origin-aggregated-logging"
+    #    - name: "openshift-ansible"
+    oct sync local origin-aggregated-logging --branch $GIT_BRANCH --merge-into $GIT_BASE_BRANCH --src $GIT_REPO_BASE_DIR/origin-aggregated-logging
+    # seems to be a bug currently - doesn't checkout branch other than master - so force it to make sure
+    ssh -n openshiftdevel "cd $OS_O_A_L_DIR; git checkout $GIT_BASE_BRANCH"
+    # also doesn't handle submodules very well
+    ssh -n openshiftdevel "cd $OS_O_A_L_DIR; git submodule sync; git submodule update --init --recursive --remote"
+    #oct sync remote openshift-ansible --branch master
+    oct sync local openshift-ansible --branch $ANSIBLE_BRANCH --merge-into $ANSIBLE_BASE_BRANCH --src $GIT_REPO_BASE_DIR/openshift-ansible
+    # seems to be a bug currently - doesn't checkout branch other than master - so force it to make sure
+    ssh -n openshiftdevel "cd $OS_O_A_DIR; git checkout $ANSIBLE_BASE_BRANCH"
+    # also needs aos_cd_jobs
+    oct sync remote aos-cd-jobs --branch master
+    ip=`getremoteip`
+    fqdn=`getremotefqdn $ip`
+    SSH_HOST=openshiftdevel
+    kibana_host=kibana.$fqdn
+    kibana_ops_host=kibana-ops.$fqdn
+    update_etc_hosts $ip $fqdn $kibana_host $kibana_ops_host
 fi
-if [ "${keypair_name:-None}" = None ] ; then
-    if [ -z "${OCT_KEYPAIR_NAME:-}" ] ; then
-        echo Please set OCT_KEYPAIR_NAME to the name of the keypair you want to use
-        exit 1
-    fi
-    oct configure aws-client keypair_name "${OCT_KEYPAIR_NAME}" || {
-        echo Adding configuration for keypair_name
-        echo keypair_name: "${OCT_KEYPAIR_NAME}" >> $HOME/.config/origin-ci-tool/aws_client_configuration.yml
-    }
+if [ $PROVIDER = libvirt ] ; then
+    # assumes you have NFS mounted your /home/$USER directory from the host
+    # to /share in the VM, with root, and with root in the VM mapped to
+    # your local UID (default 1000) e.g. something like this on the host
+    # /etc/exports.d/home.exports:
+    # /home/myusername	192.168.0.0/255.255.0.0(rw,all_squash,anonuid=1000,anongid=1000)
+    # and inside the VM - /etc/fstab:
+    # 192.168.122.1:/home/myusername    /share      nfs nfsvers=3   0 0
+    # assumes you have set up the VM so that you can use your user SSH identity
+    # to login to the VM as root with no password e.g. ssh root@the.local.vm "ls"
+    # should not prompt for a password
+    TESTHOST=${TESTHOST:-$1}
+    SSH_HOST=root@$TESTHOST
+    OS_O_A_L_DIR=${OS_O_A_L_DIR:-/share/origin-aggregated-logging}
```

Comment:
We cannot rely on this default setting since if OS_O_A_L_DIR is not set in ~/.config/get-machine-run-tests.sh, the following default value is already assigned at line 146...?
`OS_O_A_L_DIR=${OS_O_A_L_DIR:-$REMOTE_GOPATH/src/github.com/openshift/origin-aggregated-logging}`

I.e., I had to have these in ~/.config/get-machine-run-tests.sh.
```
export OS_O_A_L_DIR=/share/origin-aggregated-logging
export OS_O_A_DIR=/share/openshift-ansible
export OS_A_C_J_DIR=/share/aos-cd-jobs
```